### ISSUE TITLE
Recording and sending Blob data programatically first recordedData comes undefined despite existing

### ISSUE THREAD

**Issue by 1w1111876SalasFacundo** [collab-project/videojs-record#610]
I am using the latest version of videojs-record 

So everything is great but i need the functionalities and not the default video / audio UI 
so i am trying to save the data programatically using my own custom UI 

still my problem is the next ,  when I open my modal there is a button to activate the devices then i can start and stop recording on stop recording i am saving my recorded data into an array to later sent it to my backend 

my main problem is no matter what i do the very first recording always comes as undefined instead of a Blob object , despite this I know for certain thanks to using console log along the player data that the recordedData exists and it shouldn't be coming as undefined , below is my current code .

Thanks you in advance i am open to questions 

I am using vue2 
`<template>
  <div class="qpAudioRecorderv2">
    <div>
      <v-badge
        color="success"
        :content="selectedFiles.length"
        overlap
        bordered
        :value="selectedFiles.length != 0"
      >
        <v-btn
          color="secondary primary--text"
          fab
          small
          @click="open"
        >
          <v-icon>mdi-microphone</v-icon>
        </v-btn>
      </v-badge>
    </div>
    <v-fade-transition>
      <div
        v-show="openRecorder"
        class="q-overlay q-pointer"
        :class="openRecorder ? '' : 'd-none'"
      >
        <v-container>
          <v-scale-transition>
            <div v-show="openRecorder">
              <QpCard
                v-show="openRecorder"
                rounded
                class="justify-content-center centered"
                style="width: 500px;"
              >
                <v-toolbar
                  dense
                  flat
                  rounded
                  elevation="0"
                  color="accent primary--text"
                >
                  <v-toolbar-title>{{ $t('Grabar Audio')}}</v-toolbar-title>
                  <v-spacer></v-spacer>
                  <v-btn
                    icon
                    @click="close"
                  >
                    <v-icon>mdi-window-close</v-icon>
                  </v-btn>
                </v-toolbar>
                <QpCardText>
                  <v-row justify="center">
                    <v-col
                      cols="12"
                      lg="12"
                      md="12"
                      sm="12"
                      xs="12"
                      class="d-flex justify-content-center"
                    >
                      <audio
                        id="myAudio"
                        class="video-js vjs-default-skin"
                      ></audio>
                      <div v-if="deviceEnabled === false">
                        <v-btn
                          fab
                          rounded
                          large
                          @click="enableDevice"
                        >
                          <v-icon>mdi-play</v-icon>
                        </v-btn>
                      </div>
                      <div v-if="isRecording === false && deviceEnabled === true">
                        <v-btn
                          fab
                          rounded
                          large
                          @click="beginRecord"
                        >
                          <v-icon>mdi-microphone</v-icon>
                        </v-btn>
                      </div>
                      <div v-if="isRecording === true && deviceEnabled === true">
                        <v-btn
                          fab
                          rounded
                          large
                          @click="stopRecord"
                          color="red"
                          class="recording"
                        >
                          <v-icon>mdi-microphone</v-icon>
                        </v-btn>
                      </div>
                    </v-col>
                    <v-col
                      cols="12"
                      lg="12"
                      md="12"
                      sm="12"
                      xs="12"
                      class="d-flex justify-content-center"
                    >
                      <v-btn
                        class="secondary primary--text"
                        @click="postTemplateFiles"
                      >
                        {{ $t('globalSave') }}
                      </v-btn>
                    </v-col>
                  </v-row>
                </QpCardText>
              </QpCard>
            </div>
          </v-scale-transition>
        </v-container>
      </div>
    </v-fade-transition>
  </div>
</template>

<script>
/* eslint-disable */
import QpCard from '@/components/qp/basecomponents/QpCard';
import QpCardText from '@/components/qp/basecomponents/QpCardText';
import QpButton from '@/components/qp/basecomponents/QpButton';
import QpToolbar from '@/components/qp/QpToolbar';
import 'video.js/dist/video-js.css';
import 'videojs-record/dist/css/videojs.record.css';
import videojs from 'video.js';

import 'webrtc-adapter';
import RecordRTC from 'recordrtc';

import WaveSurfer from 'wavesurfer.js';
import MicrophonePlugin from 'wavesurfer.js/dist/plugin/wavesurfer.microphone.js';
WaveSurfer.microphone = MicrophonePlugin;

// register videojs-wavesurfer plugin
import videojs_wavesurfer_css from 'videojs-wavesurfer/dist/css/videojs.wavesurfer.css';
import Wavesurfer from 'videojs-wavesurfer/dist/videojs.wavesurfer.js';
import Record from 'videojs-record/dist/videojs.record.js';

export default {
  name: 'QpAudioRecorverV2',
  components: {
    QpCard,
    QpCardText,
    QpButton,
    QpToolbar,
  },
  props: {
    objectid: { type: String, required: true },
    objecttypedata: { type: Object, required: true },
    objectstatus: { type: String, required: true },
  },
  data() {
    return {
      //QpData
      openRecorder: false,
      isRecording: false,
      deviceEnabled: false,
      selectedFiles: ['1', '2'],
      postData: new FormData(),
      //VideoJSData
      player: '',
      options: {
        controls: true,
        width: 360,
        height: 240,
        fluid: false,
        bigPlayButton: false,
        plugins: {
          wavesurfer: {
            backend: 'WebAudio',
            waveColor: '#36393b',
            progressColor: 'black',
            cursorWidth: 1,
            displayMilliseconds: true,
            hideScrollbar: true,
            plugins: [
              // enable microphone plugin
              WaveSurfer.microphone.create({
                bufferSize: 4096,
                numberOfInputChannels: 1,
                numberOfOutputChannels: 1,
                constraints: {
                  video: false,
                  audio: true,
                },
              }),
            ],
          },
          // configure videojs-record plugin
          record: {
            audio: true,
            video: false,
            maxLength: 300,
            displayMilliseconds: true,
            debug: false,
          },
        },
      },
    };
  },
  methods: {
    enableDevice() {
      this.player.record().getDevice();
      this.deviceEnabled = true;
    },
    beginRecord() {
      this.isRecording = true;
      this.player.record().start();
    },
    stopRecord() {
      this.player.record().stop();
      console.log(this.player.recordedData);
      this.selectedFiles.push(this.player.recordedData);
      this.isRecording = false;
    },
    open() {
      this.openRecorder = true;
    },
    close() {
      this.openRecorder = false;
    },
    postTemplateFiles() {
  // send data to back here
  },
  mounted() {
    /* eslint-disable no-console */
    this.player = videojs('#myAudio', this.options, () => {});
  },
};
</script>

<style scoped>
.centered {
  position: fixed;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
}

.recording {
  animation: recording-pulse 1.3s infinite ease-out;
}

@keyframes recording-pulse {
  50% {
    transform: scale(1.2);
  }
  100% {
    transform: scale(1);
  }
}
</style>`

**Comment by 1w1111876SalasFacundo** [comment]
Basically when I first click my beginRecording button , and then i click my StopRecording button the very 1st element that i push into the array comes as undefined , the next ones are BlobData that is coming as intended

**Comment by thijstriemstra** [comment]
@1w1111876SalasFacundo I cannot help you with 3rd-party frameworks like vue.js. If you can reproduce an issue with pure javascript, please add the example code and I'll take a look.

**Comment by 1w1111876SalasFacundo** [comment]
@thijstriemstra  yeah sure i will make one during the weekend and post it over here

**Comment by thijstriemstra** [comment]
no feedback, closing.
